jcl/src/java.base/share/classes/com/ibm/tools/attach/target/Attachment.java

```diff
@@ -393,13 +398,37 @@ private static String startLocalAgent() throws IbmAttachOperationFailedException
 				throw new IbmAttachOperationFailedException("startLocalManagementAgent cannot access "+START_LOCAL_MANAGEMENT_AGENT);		 //$NON-NLS-1$
 			}
 		} catch (Throwable e) {
-			throw new IbmAttachOperationFailedException("startLocalManagementAgent error starting agent:"+e.getClass()+" "+e.getMessage());		 //$NON-NLS-1$ //$NON-NLS-2$
+			throw new IbmAttachOperationFailedException("startLocalManagementAgent error starting agent:" + e.getClass()+" " + e.getMessage());		 //$NON-NLS-1$ //$NON-NLS-2$
 		}
-		String addr = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties().getProperty(IPC.LOCAL_CONNECTOR_ADDRESS);
-		if (null == addr) {
-			throw new IbmAttachOperationFailedException("startLocalManagementAgent: "+IPC.LOCAL_CONNECTOR_ADDRESS+" not defined");		 //$NON-NLS-1$ //$NON-NLS-2$
+		/*
+		 * sun.management.Agent.startLocalManagementAgent() in Java 8 sets 
+		 * c.s.m.j.localConnectorAddress in System.properties.
+		 * jdk.internal.agent.Agent.startLocalManagementAgent() in Java 9 sets 
+		 * c.s.m.j.localConnectorAddress in a different Properties object.
+		 */
+		Properties systemProperties = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties();
+		String addr;
+		/*[IF Sidecar19-SE-B165]*/
+		synchronized (systemProperties) {
+			addr = systemProperties.getProperty(LOCAL_CONNECTOR_ADDRESS);
+			if (Objects.isNull(addr)) {
+				addr = VMSupport.getAgentProperties().getProperty(LOCAL_CONNECTOR_ADDRESS);
+				/*
+				 * addr should not be null at this point.  If it is, and exception is thrown below.
+				 * Don't bother checking for null since
+				 * systemProperties had a null value before.
```

Comment:
My previous comment here was "This comment implies that calling setProperty with a null value will work, however it throws NPE." to which the response was "True. Fixed". However I don't see any change.

Reply:
Sorry, forgot to push my changes.